server/bingo/models.py

```diff
@@ -1,3 +1,73 @@
-from django.db import models # noqa
+from django.db import models
+from django.contrib.auth.models import PermissionsMixin, AbstractBaseUser, BaseUserManager
 
-# Create your models here.
+
+class UserManager(BaseUserManager):
+
+    def create_user(self, username, email, password, **extra_fields):
+
+        user = self.model(
+            username=username,
+            email=self.normalize_email(email),
+            **extra_fields
+        )
+
+        user.set_password(password)
+        user.save(using=self.db)
+        return user
+
+    def create_superuser(self, username, email, password, **extra_fields):
+        user = self.create_user(
+            username=username,
+            email=email,
+            password=password,
+            **extra_fields
+        )
+        user.is_superuser = True
+        user.save(using=self.db)
+        return user
+
+
+class User(AbstractBaseUser, PermissionsMixin):
+    REQUIRED_FIELDS = ["email"]
+
+    user_id = models.AutoField(primary_key=True)
+
+    username = models.CharField(max_length=30, unique=True)
+    USERNAME_FIELD = "username"
+
+    first_name = models.CharField(max_length=30)
+    last_name = models.CharField(max_length=30)
+    bio = models.CharField(max_length=300)
+    total_points = models.IntegerField(default=0)
+    birthdate = models.DateField(null=True)
```

Comment:
I think the ``birthdate`` field should have ``null=True`` removed and instead have some meaningless default. If you leave it as it is, you can't press save on the admin in the Django admin dashboard because a "This field is required" error pops up for the birthdate field. An alternative option is setting ``blank=True``, but I think the former option is better.

Reply:
I have enabled blank entries for emails, but I believe that in the end, blank birthdates shouldn't be allowed